dmscripts/notify_buyers_to_award_closed_briefs.py

```diff
@@ -0,0 +1,147 @@
+                    extra=logging_context
+                )
+            else:
+                logger.info(
+                    "Notifying {no_of_users} user(s) about brief ID: {brief_id} - '{brief_title}'",
+                    extra=logging_context
+                )
+                notify_client.send_email(
+                    user['emailAddress'], notify_template_id, email_context_data, allow_resend=False
+                )
+        except EmailError:
+            return user['id']
+
+
+def _log_failures(failed_users_by_brief_id, date_closed):
+    all_failed_users = []
+    for brief_id, failed_brief_users in sorted(failed_users_by_brief_id.items()):
+        logger.error(
+            'Email sending failed for the following buyer users of brief ID {brief_id}: {buyer_ids}',
+            extra={
+                "brief_id": brief_id,
+                "buyer_ids": ",".join(map(str, failed_brief_users))
+            }
+        )
+        all_failed_users.extend(failed_brief_users)
+
+    logger.error(
+        'All failures for award closed briefs notification on closing date {date_closed}: {all_failed_users}',
+        extra={
+            "date_closed": date_closed.strftime(DATE_FORMAT),
+            "all_failed_users": ",".join(map(str, all_failed_users))
+        }
+    )
+
+
+def main(
+    data_api_url, data_api_access_token, notify_api_key, notify_template_id, offset_days,
+    dry_run=None, date_closed=None, user_id_list=None
+):
+    """
+    Send emails to buyers via Notify, reminding them to award their closed briefs
+
+    offset_days:    send emails for briefs that closed X days ago
+    dry_run:        log instead of sending emails
+    date_closed:    if supplied, send emails for briefs that closed on this date
+    user_id_list:   if supplied, only send emails to buyers with these user IDs
+    """
+    logger.info("Data API URL: {data_api_url}", extra={'data_api_url': data_api_url})
+
+    date_closed = _get_brief_closing_date(offset_days, date_closed)
+    if not date_closed:
+        logger.error('Not allowed to notify about briefs that closed less than {} days ago', offset_days)
+        return False
+
+    data_api_client = dmapiclient.DataAPIClient(data_api_url, data_api_access_token)
+    notify_client = DMNotifyClient(notify_api_key, logger=logger)
+
+    closed_briefs = brief_data_helpers.get_briefs_closed_on_date(data_api_client, date_closed)
+    if not closed_briefs:
+        logger.info("No briefs closed on {date_closed}", extra={"date_closed": date_closed})
+        return True
+
+    # If user list supplied, only continue for briefs that have at least one user in that list
+    if user_id_list:
+        closed_briefs = _filter_briefs_by_user_id_list(closed_briefs, user_id_list)
+
+    logger.info("Notifying users about {briefs_count} closed brief(s)", extra={'briefs_count': len(closed_briefs)})
+
+    failed_users_by_brief_id = {}
+    for brief in closed_briefs:
+        failed_users_for_this_brief = []
```

Comment:
Yep having the headline "Notifying X users about brief Y" log message would be fine here I think (the same with or without dry-run), and then just log individual emails for the dry-run inside the `send_email...` call.

Reply:
I don't think we really need to know how many users are being emailed for each brief (yet!) so I've just taken it out. We can review this when we start having multiple brief users?

Reply:
Yep.